### PR TITLE
[WIP] Add reference table of the node types

### LINKED ISSUE #501: Where can I get the reference table of the node type?
This maybe a stupid question, but I read all the documents about this project still cannot find a full referrence table.
Currently I only know that 'Stmt' stands for 'statement' and 'Expr' means 'expression'.

### PR DESCRIPTION
**Pull request by carusogabriel**
Closes #501.

Roadmap:
- [ ] List all Nodes
- [ ] List all subNodes
- [ ] Group same types in different files
- [ ] Describe better each Node Type

### CODE REVIEW

**Comment by nikic** [comment]
[Rendered](https://github.com/carusogabriel/PHP-Parser/blob/reference-table-node-type/doc/component/node_table.markdown)

File: doc/component/node_table.markdown [hunk 1]
```diff
@@ -0,0 +1,55 @@
+## Expr
+
+Class | Node
+----- | -----
+`PhpParser\Node\Expr\BooleanNot` | `!` operator
+`PhpParser\Node\Expr\Empty_` | `empty()` function
+`PhpParser\Node\Expr\Exit_` | `exit` function
+`PhpParser\Node\Expr\Instanceof_` | `instanceof` operator
+`PhpParser\Node\Expr\Isset_` | `isset` operator
+`PhpParser\Node\Expr\List_` | `list()` language constructor. `[]` syntax is also handled by this node
```
Comment (nikic) [inline doc/component/node_table.markdown]: I believe `[] = ` syntax uses `Array_`.

**Comment by nikic** [comment]
Would it make sense to also list the subnodes (and their types)? Or would that make things too unwieldy?

File: doc/component/node_table.markdown (hunk 1 above)
Comment (carusogabriel) [reply doc/component/node_table.markdown]: Thanks, gonna fix it.

**Comment by nikic** [comment]
By subnodes I meant the properties of the node. So `Class_` has `flags`, `name`, `stmts` etc.

> Gonna try to list all of them this afternoon, so we can start to struct the files.
 
I think it would be best to keep everything in one file, otherwise search is hard. E.g. you'd already have to know whether something is an expression or a statement to search for it.

**Comment by carusogabriel** [comment]
> By subnodes I meant the properties of the node. So Class_ has flags, name, stmts etc.

Interesting didn't think on that. Added to our roadmap.

>  E.g. you'd already have to know whether something is an expression or a statement to search for it.

Good point, so let's keep this in one file for now.

I'm sorry for my delay, I'm a little busy with some stuff from the work, but I hope we can document everything this weekend :)

**Pull request closed without merging**